### PR TITLE
Add reshape method to chainer.Variable

### PR DESCRIPTION
**Pull request by yuyu2172**
This PR adds a reshape method to `chainer.Variable`.

The interface follows that of `numpy.ndarray.reshape`.

This PR is useful in a situation when you want to declare reshape at the end of a line rather than at the beginning. A simple example to demonstrate the visual difference is as follows.

```python
import numpy
import chainer
import chainer.functions as F

x = chainer.Variable(numpy.array([1,2,3,4]))

y = F.reshape(F.tile(x, 4), (4, 4))

# an alternative way to write the same thing
y = F.tile(x, 4).reshape(4, 4)
```

### CODE REVIEW

File: chainer/variable.py [hunk 1]
```diff
@@ -434,6 +434,17 @@ def add_cand(cand):
...
 
+    def reshape(self, *shape):
+        """Returns a variable of a different shape and the same content.
+
+        .. seealso::
+           :func:`chainer.functions.reshape` for full documentation,
+
+        """
+        if len(shape) == 1 and isinstance(shape[0], tuple):
+            shape = shape[0]
```
Comment (rezoo) [inline chainer/variable.py]: Could you describe why you decided to slightly change the specification of `reshape` as above?

Comment (yuyu2172) [inline chainer/variable.py]: The condition is necessary in the case when shape is passed as a tuple.

For example, the following code would raise error without the condition.

```python
import chainer
import numpy

x = chainer.Variable(numpy.array([1,2,3] ))
x.reshape((3,))
```

By the way, this API is consistent with `numpy.ndarray.reshape`.

Comment (rezoo) [inline chainer/variable.py]: Ah I overlooked. Thanks!

BTW, as this PR slightly changes API, I will ask the chainer team whether it is OK to merge or not. Please wait for a while.

Comment (rezoo) [inline chainer/variable.py]: Sorry, one comment: could you add the case where `shape[0]` is `list`?

Comment (yuyu2172) [inline chainer/variable.py]: Thanks. I fixed the code.

**Comment by rezoo** [comment]
Thank you! Then I would like to merge.

File: chainer/variable.py (hunk 1 above)
Comment (beam2d) [inline chainer/variable.py]: @yuyu2172 Would you also make it support list as well as tuple at shape[0]?

**Pull request merged**